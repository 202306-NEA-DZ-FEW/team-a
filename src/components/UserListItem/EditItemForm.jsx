import { doc, serverTimestamp, updateDoc } from "firebase/firestore";
import { useFormik } from "formik";
import Image from "next/image";
import { useTranslation } from "next-i18next";
import { useState } from "react";
import { BsImageFill } from "react-icons/bs";
import { toast } from "react-toastify";
import * as Yup from "yup";

import { db } from "@/lib/firebase";
import getAllCategories from "@/lib/getAllCategories";
import getAllStates from "@/lib/getAllStates";
import useUpdateItemImage from "@/lib/useUpdateItemImage";

import Input from "../Input";
import SelectInput from "../SelectInput";
import TextAreaInput from "../TextAreaInput";
import ImageSpinner from "../UserProfile/ImageSpinner";

function EditItemForm({ item, onEdit }) {
  const [selectedImages, setSelectedImages] = useState(item.images);
  const [imageLoading, setImageLoading] = useState(
    Array(item.images.length).fill(false)
  );

  const { updateImage } = useUpdateItemImage();
  const { t, i18n } = useTranslation();
  const states = getAllStates(t);
  const categories = getAllCategories(t);

  const formik = useFormik({
    initialValues: {
      title: item.title,
      description: item.description,
      category: item.category,
      location: item.location,
    },
    validationSchema: Yup.object({
      title: Yup.string().required(t("addItem:titleRequired")),
      description: Yup.string()
        .max(150, "description too long")
        .required(t("addItem:descriptionRequired")),
      category: Yup.string().required(t("addItem:categoryRequired")),
      location: Yup.string().required(t("addItem:locationRequired")),
    }),
    onSubmit: async (values) => {
      const itemDocRef = doc(db, "items", item.id);
      const updatedItem = {
        ...values,
        images: selectedImages,
        updated: serverTimestamp(),
      };
      try {
        await updateDoc(itemDocRef, updatedItem);
        onEdit({
          ...updatedItem,
          id: item.id,
          uid: item.uid,
          updated: serverTimestamp(),
          createdAt: item.createdAt,
          images: selectedImages,
        });
        toast.success("Profile updated successfully âœ…", {
          position: toast.POSITION.TOP_CENTER,
          autoClose: 1500,
        });
      } catch (error) {
        toast.error(error.code, {
          position: toast.POSITION.TOP_CENTER,
          autoClose: 1500,
        });
      } finally {
        document.getElementById(`my_modal_${item.id}`).close();
      }
    },
  });
  const handleImageChange = async (e, index) => {
    const imageFile = e.target.files[0];
    setImageLoading((prev) =>
      prev.map((loading, i) => (i === index ? true : loading))
    );
    const downloadURL = await updateImage(item, index, imageFile);
    setSelectedImages((prev) =>
      prev.map((image, i) => (i === index ? downloadURL : image))
    );
    setImageLoading((prev) =>
      prev.map((loading, i) => (i === index ? false : loading))
    );
  };

  return (
    <dialog
      id={`my_modal_${item.id}`}
      className='modal cursor-default bg-black bg-opacity-40'
    >
      <div
        dir={i18n.language == "ar" ? "rtl" : "ltr"}
        className='md:max-w-xl w-[90%] z-50 h-[90%]  transition-all duration-500 ease-in-out overflow-y-scroll no-scrollbar bg-base-100 rounded-2xl p-8'
      >
        <form
          onSubmit={formik.handleSubmit}
          className='flex w-full flex-col gap-3'
        >
          <h1 className='text-xl font-bold text-center'>
            {t("dashboard:editItemForm:updateItem")}
          </h1>
          <div className='flex justify-evenly gap-4 items-center'>
            {item.images.map((image, index) => (
              <figure key={index} className='relative group'>
                {imageLoading[index] && (
                  <ImageSpinner classes='absolute bg-base-100 rounded-lg w-full h-full top-0 bg-opacity-30 flex items-center justify-center' />
                )}
                <Image
                  alt={item.title}
                  width={100}
                  height={100}
                  className='w-32 h-20 object-cover rounded-lg'
                  src={selectedImages[index] || image}
                />
                <div
                  type='file'
                  className='absolute hidden text-center bg-base-100 rounded-lg w-full h-full top-0 group-hover:flex cursor-pointer bg-opacity-40 items-center justify-center'
                >
                  <input
                    type='file'
                    id={item.id + index}
                    onChange={(e) => handleImageChange(e, index)}
                    accept='image/*'
                    className='absolute hidden text-xs'
                  />
                  <label
                    htmlFor={item.id + index}
                    className='text-4xl cursor-pointer'
                  >
                    <BsImageFill />
                  </label>
                </div>
              </figure>
            ))}
          </div>
          <Input
            name='title'
            type='text'
            placeholder={t("addItem:titlePlaceholder")}
            label={t("addItem:titleLabel")}
            handleChange={formik.handleChange}
            value={formik.values.title}
            handleBlur={formik.handleBlur}
            error={formik.errors.title}
            touched={formik.touched.title}
            sm
          />
          <TextAreaInput
            name='description'
            type='text'
            placeholder={t("addItem:descriptionPlaceHolder")}
            label={t("addItem:descriptionLabel")}
            handleChange={formik.handleChange}
            value={formik.values.description}
            handleBlur={formik.handleBlur}
            error={formik.errors.description}
            touched={formik.touched.description}
            textLength={formik.values.description.length}
            maxLength={150}
          />
          <SelectInput
            name='category'
            data={categories}
            label={t("addItem:categoryLabel")}
            handleChange={formik.handleChange}
            value={formik.values.category}
            touched={formik.touched.category}
            error={formik.errors.category}
            sm
          />
          <SelectInput
            name='location'
            data={states}
            label={t("addItem:locationLabel")}
            handleChange={formik.handleChange}
            value={formik.values.location}
            touched={formik.touched.location}
            error={formik.errors.location}
            sm
          />
          <div
            className={`${i18n.language == "ar" ? "gap-2" : ""} modal-action`}
          >
            <button
              type='submit'
              className='btn rounded-xl btn-sm btn-primary bg-opacity-30 normal-case tracking-wider btn-active'
            >
              {t("common:buttons:submit")}
            </button>
            <div method='dialog'>
              <button
                type='button'
                className='btn rounded-xl btn-outline btn-sm normal-case tracking-wider font-light'
                onClick={() =>
                  document.getElementById(`my_modal_${item.id}`).close()
                }
              >
                {t("common:buttons:cancel")}
              </button>
            </div>
          </div>
        </form>
      </div>
    </dialog>
  );
}

export default EditItemForm;
