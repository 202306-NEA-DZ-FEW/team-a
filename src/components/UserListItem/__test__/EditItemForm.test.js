import renderer from "react-test-renderer";

import EditItemForm from "../EditItemForm";

jest.mock("@/lib/firebase", () => {
  return {
    app: {}, // Mock app object
    db: {}, // Mock db object
    storage: {}, // Mock storage object
    auth: {}, // Mock auth object
  };
});

it("renders correctly", () => {
  const mockOnEdit = jest.fn();
  const mockItem = {
    title: "Product Two",
    description: "Lorem ipsum description for Product Two",
    location: "djelfa",
    id: 1,
    images: [
      "https://images.unsplash.com/photo-1591047139829-d91aecb6caea?auto=format&fit=crop&q=60&w=500&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTh8fGNsb3RoZXN8ZW58MHx8MHx8fDA%3D",
      "https://images.unsplash.com/photo-1523381294911-8d3cead13475?auto=format&fit=crop&q=60&w=500&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MjB8fGNsb3RoZXN8ZW58MHx8MHx8fDA%3D",
      "https://images.unsplash.com/photo-1467043237213-65f2da53396f?auto=format&fit=crop&q=60&w=500&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8MTB8fGNsb3RoZXN8ZW58MHx8MHx8fDA%3D",
    ],
    uid: 222,
  };
  const tree = renderer
    .create(<EditItemForm item={mockItem} onEdit={mockOnEdit} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
