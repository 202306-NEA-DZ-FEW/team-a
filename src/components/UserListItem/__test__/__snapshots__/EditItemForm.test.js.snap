// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<dialog
  className="modal cursor-default bg-black bg-opacity-40"
  id="my_modal_1"
>
  <div
    className="md:w-[85%] w-[90%] z-50 h-[90%]  transition-all duration-500 ease-in-out overflow-y-scroll no-scrollbar bg-white rounded-2xl p-8"
    dir="ltr"
  >
    <form
      className="flex w-full flex-col gap-4"
      onSubmit={[Function]}
    >
      <h1
        className="text-xl font-bold text-center"
      >
        editItemForm.updateItem
      </h1>
      <div
        className="flex justify-evenly gap-4 items-center"
      >
        <figure
          className="relative group"
        >
          <img
            alt="Product Two"
            className="w-32 h-20 object-cover rounded-lg"
            data-nimg="1"
            decoding="async"
            height={100}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1591047139829-d91aecb6caea%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTh8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=256&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1591047139829-d91aecb6caea%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTh8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1591047139829-d91aecb6caea%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTh8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=256&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={100}
          />
          <div
            className="absolute hidden text-center bg-white rounded-lg w-full h-full top-0 group-hover:flex cursor-pointer bg-opacity-40 items-center justify-center"
            type="file"
          >
            <input
              accept="image/*"
              className="absolute hidden text-xs"
              id={1}
              onChange={[Function]}
              type="file"
            />
            <label
              className="text-4xl cursor-pointer"
              htmlFor={1}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M.002 3a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2h-12a2 2 0 0 1-2-2V3zm1 9v1a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V9.5l-3.777-1.947a.5.5 0 0 0-.577.093l-3.71 3.71-2.66-1.772a.5.5 0 0 0-.63.062L1.002 12zm5-6.5a1.5 1.5 0 1 0-3 0 1.5 1.5 0 0 0 3 0z"
                />
              </svg>
            </label>
          </div>
        </figure>
        <figure
          className="relative group"
        >
          <img
            alt="Product Two"
            className="w-32 h-20 object-cover rounded-lg"
            data-nimg="1"
            decoding="async"
            height={100}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1523381294911-8d3cead13475%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MjB8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=256&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1523381294911-8d3cead13475%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MjB8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1523381294911-8d3cead13475%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MjB8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=256&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={100}
          />
          <div
            className="absolute hidden text-center bg-white rounded-lg w-full h-full top-0 group-hover:flex cursor-pointer bg-opacity-40 items-center justify-center"
            type="file"
          >
            <input
              accept="image/*"
              className="absolute hidden text-xs"
              id={2}
              onChange={[Function]}
              type="file"
            />
            <label
              className="text-4xl cursor-pointer"
              htmlFor={2}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M.002 3a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2h-12a2 2 0 0 1-2-2V3zm1 9v1a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V9.5l-3.777-1.947a.5.5 0 0 0-.577.093l-3.71 3.71-2.66-1.772a.5.5 0 0 0-.63.062L1.002 12zm5-6.5a1.5 1.5 0 1 0-3 0 1.5 1.5 0 0 0 3 0z"
                />
              </svg>
            </label>
          </div>
        </figure>
        <figure
          className="relative group"
        >
          <img
            alt="Product Two"
            className="w-32 h-20 object-cover rounded-lg"
            data-nimg="1"
            decoding="async"
            height={100}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1467043237213-65f2da53396f%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTB8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=256&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1467043237213-65f2da53396f%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTB8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1467043237213-65f2da53396f%3Fauto%3Dformat%26fit%3Dcrop%26q%3D60%26w%3D500%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxzZWFyY2h8MTB8fGNsb3RoZXN8ZW58MHx8MHx8fDA%253D&w=256&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={100}
          />
          <div
            className="absolute hidden text-center bg-white rounded-lg w-full h-full top-0 group-hover:flex cursor-pointer bg-opacity-40 items-center justify-center"
            type="file"
          >
            <input
              accept="image/*"
              className="absolute hidden text-xs"
              id={3}
              onChange={[Function]}
              type="file"
            />
            <label
              className="text-4xl cursor-pointer"
              htmlFor={3}
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M.002 3a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2h-12a2 2 0 0 1-2-2V3zm1 9v1a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V9.5l-3.777-1.947a.5.5 0 0 0-.577.093l-3.71 3.71-2.66-1.772a.5.5 0 0 0-.63.062L1.002 12zm5-6.5a1.5 1.5 0 1 0-3 0 1.5 1.5 0 0 0 3 0z"
                />
              </svg>
            </label>
          </div>
        </figure>
      </div>
      <div>
        <div
          className="label ml-2"
        >
          <label
            className="label-text"
            htmlFor="title"
          >
            titleLabel
          </label>
          <label
            className="label-text-alt "
          />
        </div>
        <input
          className="input bg-white input-primary w-full rounded-full border-opacity-25"
          id="title"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="titlePlaceholder"
          type="text"
          value="Product Two"
        />
      </div>
      <div
        className="form-control"
      >
        <div
          className="label"
        >
          <label
            className="label-text"
            htmlFor="description"
          >
            descriptionLabel
          </label>
          <label
            className="label-text-alt "
          />
        </div>
        <textarea
          className="textarea textarea-primary textarea-bordered h-24 w-full bg-white"
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="descriptionPlaceHolder"
          type="text"
          value="Lorem ipsum description for Product Two"
        />
        <div
          className="label"
        >
          <label
            className="label-text"
            htmlFor="description"
          />
          <label
            className="label-text-alt "
          >
            39
            /
            150
          </label>
        </div>
      </div>
      <div>
        <div
          className="label"
        >
          <label
            className="label-text"
            htmlFor="category"
          >
            categoryLabel
          </label>
          <label
            className="label-text-alt "
          />
        </div>
        <select
          className="select select-primary bg-white w-full border-opacity-25 rounded-3xl"
          id="category"
          onChange={[Function]}
        >
          <option
            value="babyItems"
          >
            babyItems
          </option>
          <option
            value="blanketsAndBedding"
          >
            blanketsAndBedding
          </option>
          <option
            value="books"
          >
            books
          </option>
          <option
            value="clothes"
          >
            clothes
          </option>
          <option
            value="accessories"
          >
            accessories
          </option>
          <option
            value="craftSupplies"
          >
            craftSupplies
          </option>
          <option
            value="decorAndArtwork"
          >
            decorAndArtwork
          </option>
          <option
            value="electronics"
          >
            electronics
          </option>
          <option
            value="foodAndNonPerishables"
          >
            foodAndNonPerishables
          </option>
          <option
            value="householdItems"
          >
            householdItems
          </option>
          <option
            value="hygieneProducts"
          >
            hygieneProducts
          </option>
          <option
            value="medicalSupplies"
          >
            medicalSupplies
          </option>
          <option
            value="petSupplies"
          >
            petSupplies
          </option>
          <option
            value="schoolAndEducationalSupplies"
          >
            schoolAndEducationalSupplies
          </option>
          <option
            value="shoes"
          >
            shoes
          </option>
          <option
            value="sportsEquipment"
          >
            sportsEquipment
          </option>
          <option
            value="toysAndGames"
          >
            toysAndGames
          </option>
        </select>
      </div>
      <div>
        <div
          className="label"
        >
          <label
            className="label-text"
            htmlFor="location"
          >
            locationLabel
          </label>
          <label
            className="label-text-alt "
          />
        </div>
        <select
          className="select select-primary bg-white w-full border-opacity-25 rounded-3xl"
          defaultValue="djelfa"
          id="location"
          onChange={[Function]}
        >
          <option
            value="adrar"
          >
            adrar
          </option>
          <option
            value="chlef"
          >
            chlef
          </option>
          <option
            value="laghouat"
          >
            laghouat
          </option>
          <option
            value="oumElBouaghi"
          >
            oumElBouaghi
          </option>
          <option
            value="batna"
          >
            batna
          </option>
          <option
            value="bejaia"
          >
            bejaia
          </option>
          <option
            value="biskra"
          >
            biskra
          </option>
          <option
            value="bechar"
          >
            bechar
          </option>
          <option
            value="blida"
          >
            blida
          </option>
          <option
            value="bouira"
          >
            bouira
          </option>
          <option
            value="tamanrasset"
          >
            tamanrasset
          </option>
          <option
            value="tebessa"
          >
            tebessa
          </option>
          <option
            value="tlemcen"
          >
            tlemcen
          </option>
          <option
            value="tiaret"
          >
            tiaret
          </option>
          <option
            value="tiziOuzou"
          >
            tiziOuzou
          </option>
          <option
            value="algiers"
          >
            algiers
          </option>
          <option
            value="djelfa"
          >
            djelfa
          </option>
          <option
            value="jijel"
          >
            jijel
          </option>
          <option
            value="setif"
          >
            setif
          </option>
          <option
            value="saida"
          >
            saida
          </option>
          <option
            value="skikda"
          >
            skikda
          </option>
          <option
            value="sidiBelAbbes"
          >
            sidiBelAbbes
          </option>
          <option
            value="annaba"
          >
            annaba
          </option>
          <option
            value="guelma"
          >
            guelma
          </option>
          <option
            value="constantine"
          >
            constantine
          </option>
          <option
            value="medea"
          >
            medea
          </option>
          <option
            value="mostaganem"
          >
            mostaganem
          </option>
          <option
            value="mSila"
          >
            mSila
          </option>
          <option
            value="mascara"
          >
            mascara
          </option>
          <option
            value="ouargla"
          >
            ouargla
          </option>
          <option
            value="oran"
          >
            oran
          </option>
          <option
            value="elBayadh"
          >
            elBayadh
          </option>
          <option
            value="illizi"
          >
            illizi
          </option>
          <option
            value="bordjBouArreridj"
          >
            bordjBouArreridj
          </option>
          <option
            value="boumerdes"
          >
            boumerdes
          </option>
          <option
            value="elTarf"
          >
            elTarf
          </option>
          <option
            value="tindouf"
          >
            tindouf
          </option>
          <option
            value="tissemsilt"
          >
            tissemsilt
          </option>
          <option
            value="elOued"
          >
            elOued
          </option>
          <option
            value="khenchela"
          >
            khenchela
          </option>
          <option
            value="soukAhras"
          >
            soukAhras
          </option>
          <option
            value="tipaza"
          >
            tipaza
          </option>
          <option
            value="mila"
          >
            mila
          </option>
          <option
            value="ainDefla"
          >
            ainDefla
          </option>
          <option
            value="naama"
          >
            naama
          </option>
          <option
            value="ainTemouchent"
          >
            ainTemouchent
          </option>
          <option
            value="ghardaia"
          >
            ghardaia
          </option>
          <option
            value="relizane"
          >
            relizane
          </option>
          <option
            value="timimoun"
          >
            timimoun
          </option>
          <option
            value="bordjBadjiMokhtar"
          >
            bordjBadjiMokhtar
          </option>
          <option
            value="ouledDjellal"
          >
            ouledDjellal
          </option>
          <option
            value="beniAbbes"
          >
            beniAbbes
          </option>
          <option
            value="ainSalah"
          >
            ainSalah
          </option>
          <option
            value="ainGuezzam"
          >
            ainGuezzam
          </option>
          <option
            value="touggourt"
          >
            touggourt
          </option>
          <option
            value="djanet"
          >
            djanet
          </option>
          <option
            value="elMGhair"
          >
            elMGhair
          </option>
          <option
            value="elMenia"
          >
            elMenia
          </option>
        </select>
      </div>
      <div
        className=" modal-action"
      >
        <button
          className="btn rounded-full normal-case tracking-wider btn-active"
          type="submit"
        >
          buttons.submit
        </button>
        <div
          method="dialog"
        >
          <button
            className="btn btn-outline rounded-full normal-case tracking-wider font-light"
            onClick={[Function]}
            type="button"
          >
            buttons.cancel
          </button>
        </div>
      </div>
    </form>
  </div>
</dialog>
`;
